components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/util/IdPManagementConstants.java

```diff
@@ -286,6 +286,8 @@
 
         public static final String DELETE_IDP_SQL = "DELETE FROM IDP WHERE (TENANT_ID=? AND NAME=?)";
 
+        public static final String DELETE_ALL_IDP_BY_TENANT_ID_SQL = "DELETE FROM IDP WHERE TENANT_ID = ?";
```

Comment:
Will all the resources related to IDP cleared from this?

Reply:
Yes. Other relevant resources also get deleted with `ON CASCADE`. The same method was followed in deleting a single IDP also.